Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/ui/checkins/SrsCheckinsFragmentViewModel.kt

```diff
@@ -0,0 +1,119 @@
+package de.rki.coronawarnapp.srs.ui.checkins
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.asLiveData
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import de.rki.coronawarnapp.presencetracing.checkins.CheckIn
+import de.rki.coronawarnapp.presencetracing.checkins.CheckInRepository
+import de.rki.coronawarnapp.presencetracing.checkins.common.completedCheckIns
+import de.rki.coronawarnapp.srs.core.model.SrsSubmissionType
+import de.rki.coronawarnapp.ui.presencetracing.attendee.checkins.consent.CheckInsConsentItem
+import de.rki.coronawarnapp.ui.presencetracing.attendee.checkins.consent.HeaderCheckInsVH
+import de.rki.coronawarnapp.ui.presencetracing.attendee.checkins.consent.SelectableCheckInVH
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.ui.SingleLiveEvent
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactory
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.combine
+import timber.log.Timber
+
+class SrsCheckinsFragmentViewModel @AssistedInject constructor(
+    @Assisted private val savedState: SavedStateHandle,
+    @Assisted private val testType: SrsSubmissionType,
+    private val checkInRepository: CheckInRepository,
+    dispatcherProvider: DispatcherProvider,
+    private val checkInsRepository: CheckInRepository
+) : CWAViewModel(dispatcherProvider) {
+
+    private val selectedSetFlow = MutableStateFlow(initialSet())
+
+    val checkIns: LiveData<List<CheckInsConsentItem>> = combine(
+        checkInRepository.completedCheckIns,
+        selectedSetFlow
+    ) { checkIns, ids ->
+        mutableListOf<CheckInsConsentItem>().apply {
+            add(headerItem(checkIns))
+            addAll(mapCheckIns(checkIns, ids))
+        }
+    }.asLiveData(context = dispatcherProvider.Default)
+
+    val events = SingleLiveEvent<SrsCheckinsNavigation>()
+
+    fun onCloseClick() {
+        Timber.d("onCancelClick")
+        events.postValue(SrsCheckinsNavigation.ShowCloseDialog)
+    }
+
+    fun goHome() {
+        Timber.d("goHome")
+        events.postValue(SrsCheckinsNavigation.GoToHome)
+    }
+
+    fun onSkipClick() {
+        Timber.d("onSkipClick")
+        events.postValue(SrsCheckinsNavigation.ShowSkipDialog)
+    }
+
+    fun onNextClick() {
+        Timber.d("onNextClick")
+        val idsWithConsent = selectedSetFlow.value
+        events.postValue(SrsCheckinsNavigation.GoToSymptomSubmission(testType, idsWithConsent.toLongArray()))
```

Comment:
you don't need to pass the ids around, it should be persisted in the repo and it is accessed from repo